log/root.go

```diff
@@ -26,8 +30,10 @@ func Root() Logger {
 }
 
 func rootLogger() *logger {
-	res, _ := root.Load().(*logger)
-	return res
+	rootMu.Lock()
```

Comment:
I use a mutex because I wanted to guarantee that concurrent calls to `log.SetDefault` wouldn't result in the slog root logger and `root` having inconsistent values.

Reply:
Imo should not be much of a problem. Should not be massively invoked